src/Api.Rest/HttpClient/OAuth/AuthenticationFlows/OidcAuthenticationFlowBase.cs

```diff
@@ -28,6 +30,38 @@ public abstract class OidcAuthenticationFlowBase
 {
 	#region methods
 
+	/// <summary>
+	/// Choose a suitable security token to use as an access token for requesting data, based on given configuration.
+	/// </summary>
+	/// <param name="tokenResponse">Response of a token request containing different security token.</param>
+	/// <param name="configuration">OAuth configuration containing the settings for authentication.</param>
+	/// <param name="expiration">Expiration time of the chosen token.</param>
+	protected static string ChooseAccessToken( TokenResponse tokenResponse, OAuthConfiguration configuration, out DateTime expiration )
+	{
+		expiration = default;
+		switch( configuration.AccessTokenType )
+		{
+			case AccessTokenType.Auto:
+				try
+				{
+					OAuthHelper.DecodeSecurityToken( tokenResponse.AccessToken );
+					expiration = DateTime.UtcNow + TimeSpan.FromSeconds( tokenResponse.ExpiresIn );
+					return tokenResponse.AccessToken;
+				}
+				catch( SecurityTokenMalformedException )
+				{
+					var claims = OAuthHelper.DecodeSecurityToken( tokenResponse.IdentityToken ).Claims;
+					expiration = OAuthHelper.ClaimsToExpirationTime( claims.ToArray() );
```

Comment:
At first, I have oriented myself on the other methods which already exist in the OAuthHelper and consume the claims. **But wait!** After checking again, further down there are the same methods using the token string. So I changed it as you suggested, since this is indeed more elegant!